streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestClient.java

```diff
@@ -16,243 +16,204 @@
  */
 package org.apache.kafka.streams.tests;
 
-import java.time.Duration;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.common.serialization.Serdes;
 import org.apache.kafka.common.utils.Bytes;
-import org.apache.kafka.streams.kstream.Consumed;
 import org.apache.kafka.streams.KafkaStreams;
 import org.apache.kafka.streams.StreamsBuilder;
 import org.apache.kafka.streams.StreamsConfig;
-import org.apache.kafka.streams.kstream.Aggregator;
+import org.apache.kafka.streams.kstream.Consumed;
 import org.apache.kafka.streams.kstream.Grouped;
-import org.apache.kafka.streams.kstream.Initializer;
 import org.apache.kafka.streams.kstream.KGroupedStream;
 import org.apache.kafka.streams.kstream.KStream;
 import org.apache.kafka.streams.kstream.KTable;
 import org.apache.kafka.streams.kstream.Materialized;
-import org.apache.kafka.streams.kstream.Predicate;
 import org.apache.kafka.streams.kstream.Produced;
 import org.apache.kafka.streams.kstream.TimeWindows;
-import org.apache.kafka.streams.kstream.ValueJoiner;
-import org.apache.kafka.streams.state.KeyValueStore;
 import org.apache.kafka.streams.state.Stores;
 import org.apache.kafka.streams.state.WindowStore;
+import org.apache.kafka.test.TestUtils;
 
+import java.time.Duration;
 import java.util.Properties;
 
 public class SmokeTestClient extends SmokeTestUtil {
 
-    private final Properties streamsProperties;
+    private final String name;
 
     private Thread thread;
     private KafkaStreams streams;
     private boolean uncaughtException = false;
+    private boolean started;
 
-    public SmokeTestClient(final Properties streamsProperties) {
+    SmokeTestClient(final String name) {
         super();
-        this.streamsProperties = streamsProperties;
+        this.name = name;
     }
 
-    public void start() {
+    public boolean started() {
+        return started;
+    }
+
+    public void start(final Properties streamsProperties) {
         streams = createKafkaStreams(streamsProperties);
-        streams.setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
-            @Override
-            public void uncaughtException(final Thread t, final Throwable e) {
-                System.out.println("SMOKE-TEST-CLIENT-EXCEPTION");
-                uncaughtException = true;
-                e.printStackTrace();
-            }
+        streams.setUncaughtExceptionHandler((t, e) -> {
+            System.out.println(name + ": SMOKE-TEST-CLIENT-EXCEPTION");
```

Comment:
add new outputs with the name, but avoid touching existing lines that are used in the system tests.

Reply:
To what extend are existing lines used? How difficult would it be to update the testa to avoid duplicate output? For example, if we only `grep` for the pattern, adding a prefix name should not require any updates to system test code.

Reply:
I agree I think that by adding a prefix with space should be fine for the test.

Reply:
:+1: I've removed the unprefixed lines, and am running the system tests to see.